Mklinks subprocess -- rcsdiff -c -rR4-Vanilla -rR4-Mvex-r1 server/dix/RCS/dispatch.c,v 
*** 1.30.1.1	1991/09/22 00:28:43
--- 1.30.1.2	1991/09/22 00:29:37
***************
*** 1294,1299 ****
--- 1294,1302 ----
      REQUEST(xCreatePixmapReq);
      DepthPtr pDepth;
      register int i;
+ #ifdef VIDEOX
+     extern Bool IsVexDepth();
+ #endif /* VIDEOX */
  
      REQUEST_SIZE_MATCH(xCreatePixmapReq);
      client->errorValue = stuff->pid;
***************
*** 1315,1320 ****
--- 1318,1327 ----
          for (i=0; i<pDraw->pScreen->numDepths; i++, pDepth++)
  	   if (pDepth->depth == stuff->depth)
                 goto CreatePmap;
+ #ifdef VIDEOX
+ 	if (IsVexDepth(pDraw->pScreen, stuff->depth))
+ 	    goto CreatePmap;
+ #endif /* VIDEOX */
  	client->errorValue = stuff->depth;
          return BadValue;
      }
***************
*** 2136,2141 ****
--- 2143,2152 ----
      ScreenPtr pScreen;
      REQUEST(xCreateColormapReq);
      int i, result;
+     Bool fOK;
+ #ifdef VIDEOX
+     extern VisualPtr GetVexVisual();
+ #endif /* VIDEOX */
  
      REQUEST_SIZE_MATCH(xCreateColormapReq);
  
***************
*** 2151,2171 ****
          return(BadWindow);
  
      pScreen = pWin->drawable.pScreen;
      for (i = 0, pVisual = pScreen->visuals;
  	 i < pScreen->numVisuals;
  	 i++, pVisual++)
      {
! 	if (pVisual->vid != stuff->visual)
! 	    continue;
! 	result =  CreateColormap(mid, pScreen, pVisual, &pmap,
! 				 (int)stuff->alloc, client->index);
! 	if (client->noClientException != Success)
! 	    return(client->noClientException);
! 	else
! 	    return(result);
      }
!     client->errorValue = stuff->visual;
!     return(BadValue);
  }
  
  int
--- 2162,2191 ----
          return(BadWindow);
  
      pScreen = pWin->drawable.pScreen;
+     fOK = FALSE;
      for (i = 0, pVisual = pScreen->visuals;
  	 i < pScreen->numVisuals;
  	 i++, pVisual++)
      {
! 	if (pVisual->vid == stuff->visual) {
! 	    fOK = TRUE;
! 	    break;
! 	}
!     }
! #ifdef VIDEOX
!     if (fOK == FALSE && (pVisual = GetVexVisual(pScreen, stuff->visual)))
! 	fOK = TRUE;
! #endif /* VIDEOX */
!     if (fOK == FALSE) {
! 	client->errorValue = stuff->visual;
! 	return(BadValue);
      }
!     result =  CreateColormap(mid, pScreen, pVisual, &pmap,
! 			     (int)stuff->alloc, client->index);
!     if (client->noClientException != Success)
! 	return(client->noClientException);
!     else
! 	return(result);
  }
  
  int
